packages/networks/src/networks/vNaga/shared/managers/api-manager/APIFactory.ts

```diff
@@ -0,0 +1,315 @@
+import { getChildLogger } from '@lit-protocol/logger';
+import { z } from 'zod';
+import {
+  EncryptedVersion1Schema,
+  GenericEncryptedPayloadSchema,
+  GenericResultBuilder,
+} from '@lit-protocol/schemas';
+import { RequestItem, NagaJitContext } from '@lit-protocol/types';
+
+import { E2EERequestManager } from './e2ee-request-manager/E2EERequestManager';
+import { composeLitUrl } from '../endpoints-manager/composeLitUrl';
+import { createRequestId } from '../../helpers/createRequestId';
+import { issueSessionFromContext } from '../session-manager/issueSessionFromContext';
+import { PricingContextSchema } from '../pricing-manager/schema';
+import { combinePKPSignSignatures } from './helper/get-signatures';
+
+import type { INetworkConfig } from '../../interfaces/NetworkContext';
+import type { PKPSignCreateRequestParams } from './pkpSign/pkpSign.CreateRequestParams';
+import type { DecryptCreateRequestParams } from './decrypt/decrypt.CreateRequestParams';
+import type { ExecuteJsCreateRequestParams } from './executeJs/executeJs.CreateRequestParams';
+
+import { PKPSignRequestDataSchema } from './pkpSign/pkpSign.RequestDataSchema';
+import { PKPSignResponseDataSchema } from './pkpSign/pkpSign.ResponseDataSchema';
+import { DecryptRequestDataSchema } from './decrypt/decrypt.RequestDataSchema';
+import { DecryptResponseDataSchema } from './decrypt/decrypt.ResponseDataSchema';
+import { ExecuteJsRequestDataSchema } from './executeJs/executeJs.RequestDataSchema';
+import { ExecuteJsResponseDataSchema } from './executeJs/executeJs.ResponseDataSchema';
+
+const _logger = getChildLogger({
+  module: 'APIFactory',
+});
+
+export function createPKPSignAPI<T, M>(networkConfig: INetworkConfig<T, M>) {
+  return {
+    createRequest: async (
+      params: PKPSignCreateRequestParams
+    ): Promise<RequestItem<z.infer<typeof EncryptedVersion1Schema>>[]> => {
+      _logger.info('pkpSign:createRequest: Creating request', { params });
+
+      // Generate session sigs
+      const sessionSigs = await issueSessionFromContext({
+        pricingContext: PricingContextSchema.parse(params.pricingContext),
+        authContext: params.authContext,
+      });
+
+      _logger.info('pkpSign:createRequest: Session sigs generated');
+
+      // Generate requests
+      const _requestId = createRequestId();
```

Comment:
why is this _requestId and the _requestData prefixed with underscore?  isn't underscore typically for unused vars?